Saule/Serialization/ResourceSerializer.cs

```diff
@@ -235,24 +248,39 @@ private JObject SerializeRelationships(ResourceGraphNode node)
 
             foreach (var kv in node.Relationships)
             {
+                var relationship = kv.Value.Relationship;
+
                 var item = new JObject();
 
                 var data = SerializeRelationshipData(kv.Value);
 
                 var relationshipId = default(string);
 
                 if (data != null
-                    && kv.Value.Relationship.Kind == RelationshipKind.BelongsTo
+                    && relationship.Kind == RelationshipKind.BelongsTo
                     && kv.Value.SourceObject != null)
                 {
                     relationshipId = (string)data["id"];
                 }
 
                 var links = new JObject();
-                AddUrl(links, "self", _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), kv.Value.Relationship));
-                AddUrl(links, "related", _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), kv.Value.Relationship, relationshipId));
+                var self = _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), relationship);
```

Comment:
Needed because I check for empty link generation. I.e unrelated to the LinkType configuration I want to avoid generating the links if the UrlBuilder returns null which I think is in line what you once wrote in https://github.com/joukevandermaas/saule/issues/121?